### PR TITLE
Sca 1516 catalog service iteration 1

### PR DESCRIPTION
**Pull request by davidhughes1-ccs**
As approved by ADSWG, wanting to merge to master.

Note that the /account calls should probably move to a shopping service common yaml - later? Or do you think I should do that now (and merge them with the oauth yaml?)

### CODE REVIEW

**Comment by tommyb82** [comment]
Quick question @davidhughes1-ccs - looks like we have moved to `snake_case` for property names in schema components here (as opposed to `camelCase` across the existing APIs.  Is that intentional / reflects the Spree API / to become the new standard for the entire TX 'layer'?

File: catalogue/catalogue-service.yaml [hunk 1]
```diff
@@ -0,0 +1,807 @@
...
+      type: http
+      scheme: bearer
+      description: |-
+        User token to authorize Cart and Checkout requests.
+
+        You can obtain it from [/spree_oauth/token](../../oauth/index.yml) endpoint.
+        It is required to associate Cart with the User.
+
+  schemas:
+    Error:
```
Comment (tommyb82) [inline catalogue/catalogue-service.yaml:256]: Should this make use of the common `Errors` type from https://github.com/Crown-Commercial-Service/ccs-scale-api-definitions/blob/SCA-1516-Catalog-Service-Iteration-1/common/CCS_Standard_API_Schema.yaml#L109 ?

Comment (davidhughes1-ccs) [reply catalogue/catalogue-service.yaml:256]: Thanks. In a perfect world yes. Initially I just copied the Spree ones (both error object and property names). Agree that the format should align with previously - even if Spree is different I'm sure you are able to translate.

Comment (davidhughes1-ccs) [reply catalogue/catalogue-service.yaml]: Have updated to use the central Error definition (and some others, AgreementID, LotID, Timestamp, Email etc). Have also corrected camelCase properties

**Review by tommyb82 — APPROVED** [review]
Looks great - if development of the API turns up any issues then we can follow up with small PRs 👍

**Pull request merged**